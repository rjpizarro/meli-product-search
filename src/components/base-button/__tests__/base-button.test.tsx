import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import BaseButton from '../index'

const mockOnClick = jest.fn()

describe('BaseButton', () => {
    it('should render correctly ', () => {
        const tree = renderer
            .create(
            <BaseButton>
                <div>children</div>
            </BaseButton>
        ).toJSON()

        const { getByTestId }  = render(
            <BaseButton>
                <div data-testid="child-component">children</div>
            </BaseButton>
        );

        expect(tree).toMatchSnapshot()
        expect(getByTestId('child-component')).toBeInTheDocument()
    })

    it('should accept any props passed', () => {
        const { getByTestId } = render(
            <BaseButton data-testid="find-me">
                <div>
                    children
                </div>
            </BaseButton>
        );

        expect(getByTestId('find-me')).toBeInTheDocument()
    })
})

