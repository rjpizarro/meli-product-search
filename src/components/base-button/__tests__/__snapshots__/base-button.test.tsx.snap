// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton should render correctly  1`] = `
<button
  className="base-button"
>
  <div>
    children
  </div>
</button>
`;
